mu-plugins/blocks/time/index.php

```diff
@@ -0,0 +1,177 @@
+ */
+function init() {
+	register_block_type( __DIR__ . '/build' );
+	add_filter('the_content', __NAMESPACE__ . '\transform_time_blocks', 99, 1);
+}
+
+/**
+ * Builds the time block output.
+ *
+ * This implements replacing the raw time strings in the post content with formatted times able to be converted to local times with JS.
+ *
+ * @param string $content Post content.
+ * @return string Content with display times reformatted.
+ */
+function transform_time_blocks( $content) {
+	if ( empty( $content ) ) {
+		return $content;
+	}
+
+	// Find the time block elements by the classname "wporg-time"
+	$dom = new \DOMDocument();
+	$dom->loadHTML( $content );
+	$xpath = new \DOMXPath( $dom );
+	$time_elements = $xpath->query( "//*[contains(concat(' ', normalize-space(@class), ' '), ' wporg-time ')]" );
+	
+	foreach ( $time_elements as $time_element ) {
+		$time_content = $time_element->nodeValue;
+		$parsed_time = parse_time( $time_content );
+
+		if ( $parsed_time === null ) {
+			continue;
+		}
+		
+		// Build the link and abbr microformat.
+		$time_element->setAttribute( 'href', 'https://www.timeanddate.com/worldclock/fixedtime.html?iso=' . gmdate( 'Ymd\THi', $parsed_time ) );
+
+		$new_time_content = $dom->createElement( 'abbr', $time_content );
+		$new_time_content->setAttribute( 'class', 'wporg-time-date' );
+		$new_time_content->setAttribute( 'title', gmdate( 'c', $parsed_time ) );
+
+		// Replace the raw time with the formatted time
+		$time_element->nodeValue = null;
+		$time_element->appendChild( $new_time_content );
+	}
+
+	$content = $dom->saveHTML();
+
+	return $content;
+}
+
+/**
+ * Parse the datetime description string and return a timestamp.
+ *
+ * @param string $content Datetime description, eg. `Monday, April 6 at 19:00 UTC`
+ * @return string Unix timestamp or null if the time string could not be parsed.
+ */
+function parse_time( $content ) {
+	// Replace non-breaking spaces with a regular white space.
+	$gmtcontent = preg_replace( '/\xC2\xA0|&nbsp;/', ' ', $content );
+	
+	// PHP understands "GMT" better than "UTC" for timezones.
+	$gmtcontent = str_replace( 'UTC', 'GMT', $gmtcontent );
+	
+	// Remove the word "at" from the string, if present. Allows strings like "Monday, April 6 at 19:00 UTC" to work.
+	$gmtcontent = str_replace( ' at ', ' ', $gmtcontent );
+	
+	// Try to parse the time, relative to the post time.
+	$time = strtotime( $gmtcontent, get_the_date( 'U' ) );
+
+	// If that didn't work, give up.
+	if ( false === $time || -1 === $time ) {
+		return null;
+	}
+
+	// Add the time converter JS code.
+	if ( ! has_action( 'wp_footer', __NAMESPACE__ . '\time_converter_script' ) ) {
+		add_action( 'wp_footer', __NAMESPACE__ . '\time_converter_script', 999 );
+	}
+
+	return $time;
+}
+
+/**
+ * Prints script to convert time in the viewers local time zone.
+ */
+function time_converter_script() {
+	?>
+	<script type="text/javascript" id="time_converter_script">
```

Comment:
Could you add this on the frontend via a `viewScript`?

Reply:
I did try that with no success, I'll have another go

Reply:
Added a `viewScript` entry but it isn't automatically enqueued. I've used something similar to [the method I used for showcase](https://github.com/WordPress/wporg-showcase-2022/blob/e6e247a49c6063c06d3414f6cd93ad9a14b402c4/source/wp-content/themes/wporg-showcase-2022/functions.php#L134-L147) for this also